From 27e1710fb2e10d2491976b4399f134efc24b604f Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Sun, 8 Jan 2017 19:03:16 +0200
Subject: [PATCH 0728/1345] dts: a80x0: add device tree configuration for
 board setup E

commit  e8dabf5b4f30f8a91cb8f2027e2aa51a77b31b48 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

A8040 DB board is a modular board,
which supports multiple possible configurations.
This patch introduce predefined board topology for setup E.

- Add device tree for E setup
	- AP_SPI(boot), AP_UART
	- CP0_I2C0, CP0_I2C1, CP0_UART1, CP0_UART2, CP0_SDIO-4
	- CP1_RGMII0, CP1_SPI, CP1_UART0

Change-Id: I2d0b2bb77ae83da42e6772367cb4a9d0d475b6cf
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35648
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35381
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 .../mvebu/a7k-a8k/armada8040-db-setup.txt          |  159 ++++++------
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts   |  272 ++++++++++++++++++++
 4 files changed, 361 insertions(+), 74 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 0a6a1e9..3819941 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -44,6 +44,9 @@ Boards:
    - "marvell,armada8040-db-D", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
+   - "marvell,armada8040-db-E", "marvell,armada8040-db"
+      when the board being used is the Armada 8040 Development board
+
    - "marvell,armada8040-db-A0", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
diff --git a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
index 21d8843..bbb4b97f 100644
--- a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
@@ -22,64 +22,65 @@ Board Setup
 For A8040 A1 revision SoC - Please use armada-8040-db-A configuration,
 			    Please use armada-8040-db-B configuration,
 			    Please use armada-8040-db-C configuration,
-			    Please use armada-8040-db-D configuration
+			    Please use armada-8040-db-D configuration,
+			    Please use armada-8040-db-E configuration
 For A8040 A0 revision SoC - Please use armada-8040-db-A0 configuration
 
-|		| 1: A			| 3: B			| 4: C			| 5: D			| 2: A0				|
-|---------------|-----------------------|-----------------------|-----------------------|-----------------------|-------------------------------|
-|Device tree	|armada-8040-db-A.dtb	|armada-8040-db-B.dtb	|armada-8040-db-C.dtb	|armada-8040-db-D.dtb	|armada-8040-db-A0.dtb		|
-|CP0-SW1:1	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP0-SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF				|
-|CP0-SW1:3	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP0-SW1:4	| OFF			| OFF			| OFF			| OFF			| OFF				|
-|CP0-SW1:5	| OFF			| OFF			| OFF			| OFF			| OFF				|
-|CP0-SW1:6	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP0-SW1:7	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP0-SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF				|
-|CP1-SW1:1	| OFF			| ON			| OFF			| OFF			| ON				|
-|CP1-SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF				|
-|CP1-SW1:3	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP1-SW1:4	| OFF			| OFF			| OFF			| OFF			| OFF				|
-|CP1-SW1:5	| ON			| ON			| ON			| ON			| OFF				|
-|CP1-SW1:6	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP1-SW1:7	| OFF			| OFF			| OFF			| OFF			| ON				|
-|CP1-SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF				|
+|		| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			| 2: A0				|
+|---------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|-------------------------------|
+|Device tree	|armada-8040-db-A.dtb	|armada-8040-db-B.dtb	|armada-8040-db-C.dtb	|armada-8040-db-D.dtb	|armada-8040-db-E.dtb	|armada-8040-db-A0.dtb		|
+|CP0-SW1:1	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP0-SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
+|CP0-SW1:3	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP0-SW1:4	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
+|CP0-SW1:5	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
+|CP0-SW1:6	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP0-SW1:7	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP0-SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
+|CP1-SW1:1	| OFF			| ON			| OFF			| OFF			| OFF			| ON				|
+|CP1-SW1:2	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
+|CP1-SW1:3	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP1-SW1:4	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
+|CP1-SW1:5	| ON			| ON			| ON			| ON			| ON			| OFF				|
+|CP1-SW1:6	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP1-SW1:7	| OFF			| OFF			| OFF			| OFF			| OFF			| ON				|
+|CP1-SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF				|
 
 DPR Setting
 ------------
-|						| 1: A		| 3: B		| 4: c				| 5: D				| 2: A0		|
-|-----------------------------------------------|---------------|---------------|-------------------------------|-------------------------------|---------------|
-| Jumper-module (CON44): SLM1564		| DPR all = 1-2	| DPR all = 1-2	| DPR 96-108 = 2-3		| DPR 96-108 = 2-3		| DPR all = 1-2	|
-| Serdes CP0 site module (CON42): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2			| DPR 2-5 = 1-2			| DPR 2-5 = 1-2	|
-| Serdes CP1 site module (CON41): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2			| DPR 2-5 = 1-2			| DPR 2-5 = 1-2	|
-| DB8040					| 		| 		|				| DPR 76,87,87 = 2-3		|		|
-|						|		|		| TDM module (CON51): SLM1448	| Audio module (CON51): SLM1449	|		|
+|						| 1: A		| 3: B		| 4: c				| 5: D				| 6: E					| 2: A0		|
+|-----------------------------------------------|---------------|---------------|-------------------------------|-------------------------------|---------------------------------------|---------------|
+| Jumper-module (CON44): SLM1564		| DPR all = 1-2	| DPR all = 1-2	| DPR 96-108 = 2-3		| DPR 96-108 = 2-3		| DPR 109-112 = 2-3, others = 1-2	| DPR all = 1-2	|
+| Serdes CP0 site module (CON42): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2			| DPR 2-5 = 1-2			| DPR 2-5 = 1-2				| DPR 2-5 = 1-2	|
+| Serdes CP1 site module (CON41): SLM1521	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2	| DPR 2-5 = 1-2			| DPR 2-5 = 1-2			| DPR 2-5 = 1-2				| DPR 2-5 = 1-2	|
+| DB8040					| 		| 		|				| DPR 76,87,87 = 2-3		| DPR 81 = 2-3, DPR 119,121,123-126 =1-2|		|
+|						|		|		| TDM module (CON51): SLM1448	| Audio module (CON51): SLM1449	|					|		|
 
 The tables below summarizes the interface configuration of each setup
 
 SerDes Configuration
 ------------------------
 
- CP0 Lane	| 1: A			| 3: B			| 4: C			| 5: D			| 2: A0			|
-----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
- 0		| PCIe0 (x1)		| PCIe0 (x4)		| PCIe0 (x1)		| PCIe0 (x1)		| SGMII2 3.125G		|
- 1		| SATA0			| PCIe0 (x4)		| SATA0			| SATA0			| USB3_HOST0		|
- 2		| KR (10G)		| PCIe0 (x4)		| KR (10G)		| KR (10G)		| SGMII0		|
- 3		| SATA1			| PCIe0 (x4)		| SATA1			| SATA1			| SATA1			|
- 4		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		|
- 5		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		|
+ CP0 Lane	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			| 2: A0			|
+----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
+ 0		| PCIe0 (x1)		| PCIe0 (x4)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| SGMII2 3.125G		|
+ 1		| SATA0			| PCIe0 (x4)		| SATA0			| SATA0			| SATA0			| USB3_HOST0		|
+ 2		| KR (10G)		| PCIe0 (x4)		| KR (10G)		| KR (10G)		| KR (10G)		| SGMII0		|
+ 3		| SATA1			| PCIe0 (x4)		| SATA1			| SATA1			| SATA1			| SATA1			|
+ 4		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		| USB3_HOST1		|
+ 5		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		|
 
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
 
- CP1 Lane	| 1: A			| 3: B			| 4: C			| 5: D			| 2: A0			|
-----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
- 0		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| SGMII2 3.125G		|
- 1		| SATA0			| USB3_HOST0		| SATA0			| SATA0			| USB23_HOST0		|
- 2		| KR (10G)		| SGMII0		| KR (10G)		| KR (10G)		| KR (10G)		|
- 3		| SATA1			| SATA1			| SATA1			| SATA1			| SATA1			|
- 4		| PCIe1 (x1)		| PCIe1 (x1)		| PCIe1 (x1)		| PCIe1 (x1)		| N/C			|
- 5		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		|
+ CP1 Lane	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			| 2: A0			|
+----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
+ 0		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| PCIe0 (x1)		| SGMII2 3.125G		|
+ 1		| SATA0			| USB3_HOST0		| SATA0			| SATA0			| SATA0			| USB23_HOST0		|
+ 2		| KR (10G)		| SGMII0		| KR (10G)		| KR (10G)		| KR (10G)		| KR (10G)		|
+ 3		| SATA1			| SATA1			| SATA1			| SATA1			| SATA1			| SATA1			|
+ 4		| PCIe1 (x1)		| PCIe1 (x1)		| PCIe1 (x1)		| PCIe1 (x1)		| PCIe1 (x1)		| N/C			|
+ 5		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		| PCIe2 (x1)		|
 
 - USB2_H0 (UTMI only)
 
@@ -87,40 +88,43 @@ SerDes Configuration
 Multi-purpose pin configurations
 --------------------------------
 
- AP806 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 2: A0			|
-----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
-    AP-SDIO	| [0-10]		| [0-10]		| [0-10]		| [0-10]		| N/C			|
-    I2C0	| N/C			| N/C			| N/C			| N/C			| [4-5]			|
-    AP-SPI0	| N/C			| N/C			| N/C			| N/C			| [0,3]			|
-    AP-UART0	| [11,19]		| [11,19]		| [11,19]		| [11,19]		| [11,19]		|
-
- CP0 pin	| 1: A 			| 3: B 			| 4: C			| 5: D			| 2: A0			|
-----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
-    CP0-SMI	| [32,34]		| [32,34]		| [32,34]		| [32,34]		| [32,34]		|
-    CP0-XSMI	| [42-43]		| [42-43]		| [42-43]		| [42-43]		| [35-36]		|
-    CP0-TWSI	| [37-38]		| [37-38]		| [37-38]		| [37-38]		| [37-38]		|
-    CP0-SATA1	| [40]			| [40]			| [40]			| [40]			| [40]			|
-    CP0-SATA0	| [41]			| [41]			| [41]			| [41]			| [41]			|
-    CP0-RGMII1	| [44-55]		| [44-55]		| [44-55]		| [44-55]		| [44-55]		|
-    CP0-SD	| [56-62]		| [56-62]		| [56-62]		| [56-62]		| [56-62]		|
-    CP0-PTP_CLK	| [39]			| [39]			| [39]			| [39]			| N/C			|
+ AP806 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			| 2: A0			|
+----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
+    AP-SDIO	| [0-10]		| [0-10]		| [0-10]		| [0-10]		| N/C			| N/C			|
+    I2C0	| N/C			| N/C			| N/C			| N/C			| N/C			| [4-5]			|
+    AP-SPI0	| N/C			| N/C			| N/C			| N/C			| [0,3]			| [0,3]			|
+    AP-UART0	| [11,19]		| [11,19]		| [11,19]		| [11,19]		| [11,19]		| [11,19]		|
+
+ CP0 pin	| 1: A 			| 3: B 			| 4: C			| 5: D			| 6: E			| 2: A0			|
+----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
+    CP0-SMI	| [32,34]		| [32,34]		| [32,34]		| [32,34]		| N/C			| [32,34]		|
+    CP0-XSMI	| [42-43]		| [42-43]		| [42-43]		| [42-43]		| [42-43]		| [35-36]		|
+    CP0-TWSI	| [37-38]		| [37-38]		| [37-38]		| [37-38]		| [37-38]		| [37-38]		|
+    CP0-SATA1	| [40]			| [40]			| [40]			| [40]			| [40]			| [40]			|
+    CP0-SATA0	| [41]			| [41]			| [41]			| [41]			| [41]			| [41]			|
+    CP0-RGMII1	| [44-55]		| [44-55]		| [44-55]		| [44-55]		| N/C			| [44-55]		|
+    CP0-SD	| [56-62]		| [56-62]		| [56-62]		| [56-62]		| [56-62]		| [56-62]		|
+    CP0-PTP_CLK	| [39]			| [39]			| [39]			| [39]			| [39]			| N/C			|
+    CP0-TWSI1	| N/C			| N/C			| N/C			| N/C			| [35-36]		| N/C			|
+    CP0-UART1	| N/C			| N/C			| N/C			| N/C			| [47,49]		| N/C			|
+    CP0-UART2	| N/C			| N/C			| N/C			| N/C			| [50-51]		| N/C			|
 
 GPIOS:
 	- 33: GPIO: GE_INT#/push button/Wake
-	- 35: MSS_GPIO[3]: MSS_PWDN (only in configuration 1,4,5)
-	- 36: MSS_GPIO[5]: MSS_VTT_EN (only in configuration 1,4,5)
-
- CP1 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 2: A0			|
-----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
-    CP1-RGMII0	| [0-11]		| [0-11]		| N/C			| N/C			| [0-11]		|
-    CP1-SMI	| [27,31]		| [27,31]		| N/C			| N/C			| [27,31]		|
-    CP1-SPI1	| [13-16]		| [13-16]		| [13-16]		| N/C			| N/C			|
-    CP1-UART0	| [29-30]		| [29-30]		| [29-30]		| [29-30]		| N/C			|
-    CP1-SATA1	| [28]			| [28]			| [28]			| [28]			| N/C			|
-    CP1-SATA0	| N/C			| N/C			| [27]			| N/C			| N/C			|
-    CP1-AU	| N/C			| N/C			| N/C			| [0-5]			| N/C			|
-    CP1-NAND	| N/C			| N/C			| N/C			| [13,15-27]		| N/C			|
-    CP1-TDM	| N/C			| N/C			| [0-11]		| N/C			| N/C			|
+	- 35: MSS_GPIO[3]: MSS_PWDN (only in configuration 1,4,5,6)
+	- 36: MSS_GPIO[5]: MSS_VTT_EN (only in configuration 1,4,5,6)
+
+ CP1 pin	| 1: A			| 3: B			| 4: C			| 5: D			| 6: E			| 2: A0			|
+----------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|-----------------------|
+    CP1-RGMII0	| [0-11]		| [0-11]		| N/C			| N/C			| [0-11]		| [0-11]		|
+    CP1-SMI	| [27,31]		| [27,31]		| N/C			| N/C			| [27,31]		| [27,31]		|
+    CP1-SPI1	| [13-16]		| [13-16]		| [13-16]		| N/C			| [13-16]		| N/C			|
+    CP1-UART0	| [29-30]		| [29-30]		| [29-30]		| [29-30]		| [29-30]		| N/C			|
+    CP1-SATA1	| [28]			| [28]			| [28]			| [28]			| [28]			| N/C			|
+    CP1-SATA0	| N/C			| N/C			| [27]			| N/C			| N/C			| N/C			|
+    CP1-AU	| N/C			| N/C			| N/C			| [0-5]			| N/C			| N/C			|
+    CP1-NAND	| N/C			| N/C			| N/C			| [13,15-27]		| N/C			| N/C			|
+    CP1-TDM	| N/C			| N/C			| [0-11]		| N/C			| N/C			| N/C			|
 
 Network configuration
 ---------------------
@@ -153,6 +157,13 @@ Network configuration
  eth1		| 0	| 2		| 3		| RGMII1			|
  eth2		| 1	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
 
+6: E:
+ Interface	| CPN#	| PPv2 Port	| GMAC Port	| Board Interface		|
+ ---------------|-------|---------------|---------------|-------------------------------|
+ eth0		| 0	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
+ eth1		| 1	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
+ eth2		| 1	| 1		| 2		| RGMII0			|
+
 2: A0:
  Interface	| CPN#	| PPv2 Port	| GMAC Port	| Board Interface		|
  ---------------|-------|---------------|---------------|-------------------------------|
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 76d23d84..304cf58 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -21,6 +21,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-E.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A0.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-80x0-customer.dtb
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
new file mode 100644
index 0000000..4edb1e4
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -0,0 +1,272 @@
+/*
+ * Copyright (C) 2017 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 8040 development board
+ * This file supports option E configuration
+ * CP1_RGMII0,AP_I2C,CP0_I2C0,CP0_I2C1,CP1_SPI,AP_SPI,4xUART(AP,CP0_UA1,2xCP1_UA0),CP0_SDIO-4b
+ */
+
+#include "armada-8040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-8040 development board E setup";
+	compatible = "marvell,armada8040-db-E", "marvell,armada8040-db", "marvell,armada8040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			serial@512000 {
+				status = "okay";
+			};
+			sdhci@6e0000 {
+				status = "disabled";
+			};
+			i2c@511000 {
+				status = "disabled";
+			};
+			spi0: spi@510600 {
+				status = "okay";
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			sata@540000 {
+				status = "okay";
+				sata-port@0 {
+					phys = <&comphy0 1 COMPHY_SATA0>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			serial@702100 {
+				status = "okay";
+			};
+			serial@702200 {
+				status = "okay";
+			};
+			/* usb3h0 is USB2 only (UTMI) */
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			i2c@701000 {
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_i2c0_pins>;
+				status = "okay";
+			};
+			i2c@701100 {
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_i2c1_pins>;
+				status = "okay";
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "none"; /* disabled */
+				};
+				emac3: mac3 {
+					phy-mode = "none"; /* disabled */
+				};
+			};
+			ppv22@000000 {
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+	cpn-110-slave {
+		config-space {
+			/* usb3h0_1 is USB2 only (UTMI) */
+			usb3h0_1: usb3@500000 {
+				status = "okay";
+			};
+			serial@702000 {
+				status = "okay";
+			};
+			sata@540000 {
+				status = "okay";
+				sata-port@0 {
+					phys = <&comphy1 1 COMPHY_SATA0>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+				sata-port@1 {
+					phys = <&comphy1 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				mdio1_phy1: ethernet-phy@1 {
+					reg = <0>;
+				};
+			};
+			gop {
+				emac0_1: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2_1: mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&mdio1_phy1>;
+				};
+				emac3_1: mac3 {
+					phy-mode = "none"; /* disabled */
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_ge1_rgmii_pins>;
+
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "okay";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			comphy {
+				status = "okay";
+			};
+			spi1: spi@700680 {
+				status = "okay";
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_spi1_pins>;
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x620000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
-- 
1.7.9.5

